### DISCUSSION TITLE
reusing a resource table in a relation manager - "create" button not showing up

### DISCUSSION THREAD

**Discussion by jdion84** [filamentphp/filament#9512] [Help]
i am trying to just re-use the resource table in a relation manager

e.g.:

```php
namespace App\Filament\RelationManagers;

use App\Filament\Resources\CommResource;
use Filament\Forms\Form;
use Filament\Resources\RelationManagers\RelationManager;
use Filament\Tables\Table;

class CommsRelationManager extends RelationManager
{
    protected static string $relationship = 'comms';

    public function form(Form $form): Form
    {
        return CommResource::form($form);
    }

    public function table(Table $table): Table
    {
        return CommResource::table($table);
    }
}
```

this works amazing, except the `CreateAction` button does not show up in the relation manager.

i have added the `CreateAction` to my `CommResource`  table `headerActions`:

```php
->headerActions([
    Tables\Actions\CreateAction::make(),
])
```

it still will not show up in the relation manager table. how do i make it so the create button will appear in relation manager tables?

**Comment by leandrocfe** [comment]
```php
public function table(Table $table): Table
{
    return CommResource::table($table)
         ->headerActions([
            Tables\Actions\CreateAction::make()
        ]);
}
```
?

**Reply by jdion84** [reply]
that is already in the `CommResource::table` though as i mentioned in OP

**Reply by leandrocfe** [reply]
this was supposed to work. Are you using policies?

**Reply by jdion84** [reply]
yes, is there a special auth ability i need to consider for relation manager authorizations in regards to creating hasmany relations?

**Reply by leandrocfe** [reply]
you should add the `->headerActions()` on the CommsRelationManager class

**Reply by leandrocfe** [reply]
Did you do [this](https://github.com/filamentphp/filament/discussions/9512#discussioncomment-7503569) in the CommsRelationManager.php file?

**Reply by jdion84** [reply]
yes i tried your way, and it didnt work

whats weird is im using the filament export package and the export button does show in relation manager, but the "create" button does not

**Reply by leandrocfe** [reply]
Could you share the CommsRelationManager.php file?

**Reply by jdion84** [reply]
i did in the OP

**Reply by leandrocfe** [reply]
The whole code you are trying now..

**Reply by leandrocfe** [reply]
Solved?

**Reply by jdion84** [reply]
yes i am just an idiot

i was in the "view" page and the "new comm" button doesn't show up there, then i went to the "edit" page and the button was there. it makes sense that you cannot create a relation from the "view" page.

thanks for your help

**Reply by leandrocfe** [reply]
ahh ok

**Reply by mrs-kajal** [reply]
But what If I want to add new records in relation from View page?
